index.js

```diff
@@ -661,56 +978,107 @@ instance.prototype.init_presets = function () {
 		}
 	];
 
-var save;
-for (save = 0; save < 63; save++) {
-	presets.push({
-		category: 'Save Preset',
-		label: 'Save Preset '+ parseInt(save+1) ,
-		bank: {
-			style: 'text',
-			text: 'SAVE\\nPSET\\n' + parseInt(save+1) ,
-			size: '14',
-			color: '16777215',
-			bgcolor: self.rgb(0,0,0),
-		},
-		actions: [
-			{
-				action: 'savePset',
-				options: {
-				val: ('0' + save.toString(16).toUpperCase()).substr(-2,2),
+	var save;
+	for (save = 0; save < 64; save++) {
+		presets.push({
+			category: 'Save Preset',
+			label: 'Save Preset '+ parseInt(save) ,
+			bank: {
+				style: 'text',
+				text: 'SAVE\\nPSET\\n' + parseInt(save) ,
+				size: '14',
+				color: '16777215',
+				bgcolor: self.rgb(0,0,0),
+			},
+			actions: [
+				{
+					action: 'savePset',
+					options: {
+						val: save.toString(16).padStart(2,'0').toUpperCase(),
+					}
 				}
-			}
-		]
-	});
-}
+			]
+		});
+	}
 
-var recall;
-for (recall = 0; recall < 63; recall++) {
-	presets.push({
-		category: 'Recall Preset',
-		label: 'Recall Preset '+ parseInt(recall+1) ,
-		bank: {
-			style: 'text',
-			text: 'Recall\\nPSET\\n' + parseInt(recall+1) ,
-			size: '14',
-			color: '16777215',
-			bgcolor: self.rgb(0,0,0),
-		},
-		actions: [
-			{
-				action: 'recallPset',
-				options: {
-				val: ('0' + recall.toString(16).toUpperCase()).substr(-2,2),
+	var recall;
+	for (recall = 0; recall < 64; recall++) {
```

Comment:
There seems to be a disconnect here because the loop still starts at 0 but ends +1 at 64.  Below the offsets for id 0 = lexical 1 have been removed.  The offset is no longer compensated for.

Reply:
Same applies to the save presets loop

Reply:
Let me see if I can reiterate what you are saying to make sure I understand what you're saying. In the old way it started at 0 and ended at 62 (`recall < 63`). The label, however, used to to be one more than the `val`. So, it would be something like this:

```
text: Recall PSET 1
val:  00
```
With the proposed changes it would be the following:

```
text: Recall PSET 0
val:  00
```

The reason for this change is to be more accurate. The cameras support a preset 0, but in the old way it appeared there was no way to set preset 0 (you had to set what looked like preset 1 in order to set true preset 0). Preset 0 is a special preset where the camera will automatically move to that preset upon power up.

Increasing the loop test from `recall < 63` to `recall < 64` will create presets for 0-63. Before it would only create presets for 0-62 (but labels 1-63).

Does that answer your question?